linode/manager
# change: [M3-7454] - Add child access user permissions for parent accounts

## Description 📝
As an admin on a parent account, when I create new non-admin users, I need the ability to grant full account access to child accounts. To facilitate this, a new permissions toggle labeled "Enable child account access" will be introduced during the creation of non-admin parent account users. Additionally, the interface will be enhanced by incorporating a "General permissions" headline, and the "Full account access" option will be relocated within the Paper section for improved clarity.

Also, I named the ticket category `change` because although the 'Enable Child Access' toggle button is a future feature, there are a couple UI changes that will take place as soon as this ticket is merged.

## Changes  🔄
- New "Enable child account access" toggle during the creation of non-admin parent account users.
- Introduce a new headline titled "General permissions" to provide clarity and organization.
- Relocate the "Full account access" option within the Paper section for improved visibility and alignment with the overall interface design.
- Remove the borders between each permission toggle
- Invalidate query when permissions change
- Improved mobile view

## Preview 📷
| Before  | After   |
| ------- | ------- |
| ![Screenshot 2023-12-19 at 1 52 10 PM](https://github.com/linode/manager/assets/139489745/1dc33cf0-7800-4237-91b5-5c48c8319d72) | ![Screenshot 2023-12-19 at 1 47 06 PM](https://github.com/linode/manager/assets/139489745/ca25d975-3bea-4bf7-955b-969840dd7b1c) |
| ![Screenshot 2023-12-19 at 1 52 19 PM](https://github.com/linode/manager/assets/139489745/91aaf654-cc69-4fc4-92d7-bbf6cb8dc2f2) | ![Screenshot 2023-12-19 at 1 49 30 PM](https://github.com/linode/manager/assets/139489745/234ce436-3c78-4218-a880-34780303327c) |

## How to test 🧪

### Prerequisites
- Use the Mock Service Worker
- Ensure Parent/Child Account Feature Flag is on

### Verification steps 
(How to verify changes)
- Navigate to http://localhost:3000/account/users/User/permissions
- Ensure the toggle works and displays the permissions
- Ensure there is an 'Enable child account access' toggle
- Ensure there is _no_ toggle for http://localhost:3000/account/users/ChildUser/permissions and http://localhost:3000/account/users/ParentCompany_a1b2c3d4e5/permissions because they are child and proxy accounts
- Ensure there is no 'Enable child account access' toggle button when the feature flag is off

## As an Author I have considered 🤔

*Check all that apply*
